src/RequestsLibrary/log.py

```diff
@@ -28,9 +29,12 @@ def log_request(
     args.pop('session', None)
     # This will log specific headers merged with session defined headers
     merged_headers = utils.merge_headers(session, args.pop('headers', None))
-    formatted_data = utils.format_data_to_log_string_according_to_headers(session,
-                                                                          args.pop('data', None),
-                                                                          merged_headers)
+    data = args.pop('data', None)
+    if data and not isinstance(data, io.IOBase):
```

Comment:
I would move this check inside `utils.format_data_to_log_string_according_to_headers`